### PR TITLE
ED-4515: Cantina Audit 9 - Extend time before proving new settlement world state

### PR DESCRIPTION
**Pull request by johnwhitton**

### CODE REVIEW

File: contracts/Prover.sol [hunk 1]
```diff
@@ -32,6 +32,9 @@ contract Prover is SimpleProver {
     // Output slot for the game status (fixed)
     uint256 public constant L2_FAULT_DISPUTE_GAME_STATUS_SLOT = 0;
 
+    // Number of blocks to wait before Settlemnent Layer can be proven again
```
Comment (nknavkal) [inline contracts/Prover.sol:35]: nit: typo on 'settlement'

File: contracts/Prover.sol [hunk 2]
```diff
@@ -106,6 +109,13 @@ contract Prover is SimpleProver {
         uint256 indexed _destinationChainID, uint256 indexed _blockNumber, bytes32 _L2WorldStateRoot
     );
 
+    /**
+     * @notice emitted on a proving state if the blockNumber is less than or equal to the current blockNumber + SETTLEMENT_BLOCKS_DELAY
+     * @param _inputBlockNumber the block number we are trying to prove
+     * @param _neededBlockNumber the  next block number that can been proven
```
Comment (nknavkal) [inline contracts/Prover.sol:115]: SUPER ULTRA MEGA NIT: extra space between 'the' and 'next' on line 115

File: contracts/Prover.sol [hunk 3]
```diff
@@ -287,11 +297,11 @@ contract Prover is SimpleProver {
             stateRoot: bytes32(RLPReader.readBytes(RLPReader.readList(rlpEncodedBlockData)[3]))
         });
         BlockProof memory existingBlockProof = provenStates[settlementChainId];
-        if (existingBlockProof.blockNumber < blockProof.blockNumber) {
+        if ((existingBlockProof.blockNumber + SETTLEMENT_BLOCKS_DELAY) < blockProof.blockNumber) {
```
Comment (nknavkal) [inline contracts/Prover.sol:300]: are these parentheses necessary? not sure but thought id call it out. same on line 304

**Review by nknavkal — APPROVED** [review]
bunch of stupid nits i apologize in advance but ya good to go

File: contracts/Prover.sol (hunk 2 above)
Comment (nknavkal) [reply contracts/Prover.sol:115]: more nits: 'that can been proven' --> 'that can be proven'
also i dont love the name of this param, can we do like _earliestProvableBlockNumber maybe? doesnt actually matter though

**Review by nknavkal — APPROVED** [review]
buncha nits but looks good

File: contracts/Prover.sol (hunk 1 above)
Comment (johnwhitton) [reply contracts/Prover.sol:35]: Nice catch :pray:

File: contracts/Prover.sol (hunk 2 above)
Comment (johnwhitton) [reply contracts/Prover.sol:115]: Nice catch, updated now :pray:

**Pull request merged**